django/db/models/sql/datastructures.py

```diff
@@ -166,3 +166,25 @@ def equals(self, other, with_filtered_relation):
             self.table_name == other.table_name and
             self.table_alias == other.table_alias
         )
+
+
+class BaseSubquery:
+    join_type = None
+    parent_alias = None
+    filtered_relation = None
+
+    def __init__(self, query, alias):
+        self.query = query
+        self.query.subquery = True
```

Comment:
That's the only thing that I think we should change here is too make the caller assign `subquery = True` and assert `query.subquery` here. Altering an argument is usually bad practice so I think `get_aggregation` should assign `inner_query.subquery = True` instead.